vault/resource_raft_snapshot_agent_config.go

```diff
@@ -0,0 +1,498 @@
+package vault
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-provider-vault/util"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+var snapshotAutoPath = "sys/storage/raft/snapshot-auto/config/%s"
+var allowedStorageTypes = []string{"local", "azure-blob", "aws-s3", "google-gcs"}
+
+func raftSnapshotAgentConfigResource() *schema.Resource {
+	fields := map[string]*schema.Schema{
+		"name": {
+			Type:        schema.TypeString,
+			ForceNew:    true,
+			Description: "Name of the snapshot agent configuration.",
+			Required:    true,
+		},
+		"interval_seconds": {
```

Comment:
Could we name this `interval` for consistency and parity with the Vault API?

Reply:
I'm happy to swap this, but it might be worth a quick conversation first. So this technically isn't a 1:1 mapping to the `interval` field in the Snapshot Agent API; since that field can be a Go time duration string and this is a pure integer. There are other places in this provider (`vault_mount` for instance) that append `_seconds` to the end of the field name in this case. I really don't have a strong preference, just let me know if you still want it changed.

Reply:
Ah I see. You're right, let's leave it as is :)